src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/PgQuestionAttempts.java

```diff
@@ -401,25 +401,28 @@ public void mergeAnonymousQuestionInformationWithRegisteredUserRecord(final Stri
   }
 
   @Override
-  public Map<Role, Long> getAnsweredQuestionRolesOverPrevious(final TimeInterval timeInterval)
+  public Map<TimeInterval, Map<Role, Long>> getAnsweredQuestionRolesOverPrevious(final TimeInterval[] timeRanges)
       throws SegueDatabaseException {
     String query = "SELECT role, count(DISTINCT users.id) FROM question_attempts"
         + " JOIN users ON user_id=users.id AND NOT deleted WHERE timestamp > now() - ? GROUP BY role";
-    try (Connection conn = database.getDatabaseConnection();
-         PreparedStatement pst = conn.prepareStatement(query)
-    ) {
-      pst.setObject(FIELD_GET_ANSWERED_QUESTION_ROLES_INTERVAL_AGO, timeInterval.getPGInterval());
-
-      try (ResultSet results = pst.executeQuery()) {
-        Map<Role, Long> resultsToReturn = Maps.newHashMap();
-        while (results.next()) {
-          resultsToReturn.put(Role.valueOf(results.getString("role")), results.getLong("count"));
+    Map<TimeInterval, Map<Role, Long>> allResults = new HashMap<>();
+    try (Connection conn = database.getDatabaseConnection()) {
```

Comment:
could this be refactored into some small methods to make it a bit more readable? The nested try/catches are a bit eye watering

Reply:
Changes made to refactor this into a 2 smaller methods to make it more readable